lib/utils.sh

```diff
@@ -261,20 +261,31 @@ asdf_repository_url() {
   echo "https://github.com/asdf-vm/asdf-plugins.git"
 }
 
+repository_needs_update() {
+  local update_file_dir="$(asdf_dir)/tmp"
+  local update_file_name="repo-updated"
+  # `find` outputs filename if it has not been modified in the last day
+  local find_result=$(find $update_file_dir -name "$update_file_name" -type f -mtime +1 -print)
+  [ -n "$find_result" ]
+}
+
 initialize_or_update_repository() {
   local repository_url
   local repository_path
 
   repository_url=$(asdf_repository_url)
   repository_path=$(asdf_dir)/repository
 
-  if [ -d "$repository_path" ]; then
-    echo "updating plugin repository..."
-    (cd "$repository_path" && git fetch && git reset --hard origin/master)
-  else
+  if [ ! -d "$repository_path" ]; then
```

Comment:
Why this change? The previous if condition was better because it didn't negate the directory check.

Reply:
I wanted to avoid fetching the repository every time.
With the new condition, repository will only be fetched once a day. 
I do not think to be neglecting the directory check here, what do you mean?

Reply:
My bad, I missed the new `elif` below.